File: server/src/IssueTracker.Domain/Models/Project.cs
using IssueTracker.Domain.Models.Enums;

namespace IssueTracker.Domain.Models;

public record Project 
{
    public string Id { get; set; }
    public string Summary { get; set; }
    public string Description { get; set; }
    public string CreatedBy { get; set; }
    public string OwnedBy { get; set; }
    public DateTime CreationTime { get; set; }
    public DateTime? CompletionTime { get; set; }
    public ProjectProgress Progress { get; set; } = ProjectProgress.Open;
    public ICollection<Issue> Issues { get; set; } = new List<Issue>();
}

File: client/src/models/project/project.ts
import { z } from 'zod';
import { IssueSchema } from '@/models/issue/issue';
import { ProjectIdSchema } from '@/models/ids';
import { optionalDate } from '@/models/primitives';
import { ProjectProgressSchema } from './projectProgress';

export const ProjectSchema = z.object({
  id: ProjectIdSchema,
  summary: z.string(),
  description: z
    .string()
    .nullish()
    .transform((v) => v ?? ''),
  createdBy: z.string(),
  ownedBy: z
    .string()
    .nullish()
    .transform((v) => v ?? ''),
  progress: ProjectProgressSchema,
  creationTime: optionalDate,
  completionTime: optionalDate,
  issues: z
    .array(IssueSchema)
    .nullish()
    .transform((v) => v ?? [])
});

export type Project = z.infer<typeof ProjectSchema>;
